### PR TITLE
Add personal keymaps

### PR DESCRIPTION
**Pull request by tucznak**
## Description

Adding personal keymaps for some KBDfans boards to the project. KBD67 and KBD75 aren't really interesting, NIU Mini contains a complex layout that should be very useful for any ortho users using Czech keymap.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/kbdfans/kbd67/rev2/keymaps/tucznak/rules.mk [hunk 1]
```diff
@@ -0,0 +1,82 @@
+# MCU name
```
Comment (noroadsleft) [inline keyboards/kbdfans/kbd67/rev2/keymaps/tucznak/rules.mk]: This file only needs to contain the rules that you have changed from the keyboard-level `rules.mk` file (`keyboards/kbdfans/kbd67/rev2/rules.mk`).

This entire file can be collapsed down to:

```makefile
MOUSEKEY_ENABLE = no         # Mouse keys(+4700)
CONSOLE_ENABLE = no          # Console for debug(+400)
NKRO_ENABLE = yes            # USB Nkey Rollover
```

Though if you want, you could collapse it down to only the Build Options: https://github.com/qmk/qmk_firmware/blob/c296af50a3a7a87864988fdb42e1432a2abc4057/keyboards/kbdfans/kbd67/rev2/keymaps/tucznak/rules.mk#L61-L80

This has the slight advantage of keeping your options the same in the event anything changes at the keyboard level.

File: keyboards/kbdfans/kbd75/keymaps/tucznak/config.h [hunk 2]
```diff
@@ -0,0 +1,46 @@
+#ifndef CONFIG_USER_H
```
Comment (noroadsleft) [inline keyboards/kbdfans/kbd75/keymaps/tucznak/config.h]: This file should be changed to use the `#pragma once` include guard.

```suggestion
#pragma once
```

File: keyboards/kbdfans/kbd75/keymaps/tucznak/config.h [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+#define CONFIG_USER_H
```
Comment (noroadsleft) [inline keyboards/kbdfans/kbd75/keymaps/tucznak/config.h]:
```suggestion
```

File: keyboards/kbdfans/kbd75/keymaps/tucznak/config.h [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+#include "../../config.h"
```
Comment (noroadsleft) [inline keyboards/kbdfans/kbd75/keymaps/tucznak/config.h]: QMK's build script automatically includes all the `config.h` files in the build tree, so Line 4 is redundant.

```suggestion
```

File: keyboards/kbdfans/kbd75/keymaps/tucznak/config.h [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+/* turn off RGB when computer sleeps */
+#ifdef RGB_DI_PIN
+#define RGBLIGHT_SLEEP
+#endif
+
+/* send tap key if no layer key was used even after tap delay */
+#define TAPPING_TERM 250
+#define RETRO_TAPPING
+
+#endif
```
Comment (noroadsleft) [inline keyboards/kbdfans/kbd75/keymaps/tucznak/config.h]:
```suggestion
```

File: keyboards/kbdfans/kbd75/keymaps/tucznak/keymap.c [hunk 6]
```diff
@@ -0,0 +1,95 @@
+#include QMK_KEYBOARD_H
+
+#define _______ KC_TRNS
```
Comment (noroadsleft) [inline keyboards/kbdfans/kbd75/keymaps/tucznak/keymap.c]: QMK defines aliases for these keycodes automatically. If you use the already-defined aliases, Lines 3 and 4 can be deleted.

https://github.com/qmk/qmk_firmware/blob/a120a000d481ed14a87988383be46c5a76604bb6/quantum/quantum_keycodes.h#L29-L31

File: keyboards/niu_mini/keymaps/tucznak/config.h [hunk 7]
```diff
@@ -0,0 +1,55 @@
+#ifndef CONFIG_USER_H
```
Comment (noroadsleft) [inline keyboards/niu_mini/keymaps/tucznak/config.h]: If you could replace the include guard (the ifndef, define, and endif at the end) with just `#pragma once`?

It's simpler, and less prone to user error, so if you could make this switch?

```suggestion
#pragma once
```

File: keyboards/niu_mini/keymaps/tucznak/config.h [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+#define CONFIG_USER_H
```
Comment (noroadsleft) [inline keyboards/niu_mini/keymaps/tucznak/config.h]:
```suggestion
```

File: keyboards/niu_mini/keymaps/tucznak/config.h [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+
+#include "../../config.h"
```
Comment (noroadsleft) [inline keyboards/niu_mini/keymaps/tucznak/config.h]:
```suggestion
```

File: keyboards/niu_mini/keymaps/tucznak/config.h [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+   - Octave shift and transpose
+   - Virtual sustain, portamento, and modulation wheel
+   - etc.
+*/
+//#define MIDI_ADVANCED
+
+/* override number of MIDI tone keycodes (each octave adds 12 keycodes and allocates 12 bytes) */
+//#define MIDI_TONE_KEYCODE_OCTAVES 2
+
+#endif
```
Comment (noroadsleft) [inline keyboards/niu_mini/keymaps/tucznak/config.h]:
```suggestion
```

File: keyboards/niu_mini/keymaps/tucznak/rules.mk [hunk 11]
```diff
@@ -0,0 +1,24 @@
...
+UNICODE_ENABLE = no             # Unicode
+BLUETOOTH_ENABLE = no           # Enable Bluetooth with the Adafruit EZ-Key HID
+RGBLIGHT_ENABLE = yes           # Enable WS2812 RGB underlight. 
+
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no           # Breathing sleep LED during USB suspend
+
+# Userspace defines
+VELOCIKEY_ENABLE=yes
+LAYOUTS = ortho_4x12 planck_mit
```
Comment (noroadsleft) [inline keyboards/niu_mini/keymaps/tucznak/rules.mk]: This only needs to be defined at the keyboard level.

```suggestion
```

File: keyboards/kbdfans/kbd75/keymaps/tucznak/keymap.c [hunk 12]
```diff
@@ -0,0 +1,95 @@
...
+	return true;
+}
+
+void led_set_user(uint8_t usb_led) {
+	if (usb_led & (1 << USB_LED_CAPS_LOCK)) {
+		DDRB |= (1 << 2); PORTB &= ~(1 << 2);
+		//rgblight_sethsv_noeeprom(145, 200, 200);
+		//rgblight_enable_noeeprom();
+	} else {
+		DDRB &= ~(1 << 2); PORTB &= ~(1 << 2);
```
Comment (fauxpark) [inline keyboards/kbdfans/kbd75/keymaps/tucznak/keymap.c]:
```suggestion
		setPinInput(B2);
		writePinLow(B2);
```

the `writePinLow()`s here are probably not needed as the PORTx bits default to 0.

**Comment by tucznak** [comment]
Implemented the suggested changes. I like to keep the else blocks even while empty for ease of on-the-fly modification, all other stuff got changed.

File: keyboards/kbdfans/kbd67/rev2/keymaps/tucznak/config.h [hunk 13]
```diff
@@ -0,0 +1,40 @@
...
+// place overrides here
+
+#define TAPPING_TERM 50
+#define RETRO_TAPPING
+
+#ifdef RGB_DI_PIN
+#define RGBLIGHT_SLEEP
+#endif
+
+#ifdef MANUFACTURER
```
Comment (drashna) [inline keyboards/kbdfans/kbd67/rev2/keymaps/tucznak/config.h]: you could remove the `ifdef`'s for these. They should always be defined.  

It's not necessary, but would make it a bit cleaner looking.

File: keyboards/kbdfans/kbd75/keymaps/tucznak/config.h [hunk 14]
```diff
@@ -0,0 +1,36 @@
+#pragma once
+
+#ifdef MANUFACTURER
```
Comment (drashna) [inline keyboards/kbdfans/kbd75/keymaps/tucznak/config.h]: As above:
you could remove the ifdef's for these. They should always be defined.

It's not necessary, but would make it a bit cleaner looking.

File: keyboards/kbdfans/kbd67/rev2/keymaps/tucznak/config.h (hunk 13 above)
Comment (tucznak) [reply keyboards/kbdfans/kbd67/rev2/keymaps/tucznak/config.h]: Okay, if I don't have to check and MANUFACTURER, PRODUCT, DESCRIPTION are always defined, I'll just straight up undef them and replace with my own.
Alternatively, not sure if this is frowned upon or not, but I could just remove the replacement of those values entirely and just stick with the default values. It was mostly a novelty anyway, but I find that Windows doesn't really display these anywhere meaningful.

**Comment by tucznak** [comment]
Removed the unneccessary `ifdef`s, generally cleaned up the code and made it a bit nicer and unified between keyboards. Sorry for being such a bother!

File: keyboards/kbdfans/kbd67/rev2/keymaps/tucznak/config.h (hunk 13 above)
Comment (drashna) [reply keyboards/kbdfans/kbd67/rev2/keymaps/tucznak/config.h]: Well, I'm doing the same thing. So ....  But yeah, I just undef. 

As for Windows, it does, if you look in the Windows 10 Settings -> Devices, or if you look in the Control Panel, under "Hardware and Sound" -> "Devices and Printers"

**Comment by drashna** [comment]
> Sorry for being such a bother!

No, no.  We're sorry for being a bother about all of the code and formatting. 


Looks good, and nice and clean now! :)

**Comment by noroadsleft** [comment]
> No, no. We're sorry for being a bother about all of the code and formatting.

:point_up_2: This.

**Pull request merged**